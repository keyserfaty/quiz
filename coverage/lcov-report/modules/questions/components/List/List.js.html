<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for modules/questions/components/List/List.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">modules/questions/components/List</a> List.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
&nbsp;
import Welcome from '../Steps/Welcome'
import Question from '../Steps/Question'
import Results from '../Steps/Results'
&nbsp;
const List = props =&gt; {
  const { step } = props
&nbsp;
  switch (step) {
    case 0: 
      return &lt;Welcome {...props} /&gt;
&nbsp;
    case 1:
      return &lt;Question {...props} /&gt;
&nbsp;
    case 2:
      return &lt;Results {...props} /&gt;
&nbsp;
    default:
      return &lt;Welcome {...props} /&gt;
  }
}
&nbsp;
export default List</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:33:11 GMT-0500 (EST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
