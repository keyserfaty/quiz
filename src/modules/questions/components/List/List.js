import React from 'react'

import Welcome from '../Steps/Welcome'
import Question from '../Steps/Question'
import Results from '../Steps/Results'

const List = props => {
  const { step } = props

  switch (step) {
    case 0: 
      return <Welcome {...props} />

    case 1:
      return <Question {...props} />

    case 2:
      return <Results {...props} />
  }
}

export default List