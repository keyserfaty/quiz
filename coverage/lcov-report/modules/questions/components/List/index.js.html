<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for modules/questions/components/List/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">modules/questions/components/List</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { isNil } from 'ramda'
&nbsp;
import List from './List'
&nbsp;
class ListContainer extends React.Component {
  constructor (props) {
    super(props)
&nbsp;
    this.state = {
      step: 0
    }
  }
&nbsp;
  onNextStep = stepNum =&gt; {
    const { step } = this.state
&nbsp;
    this.setState({
      step: !isNil(stepNum) ? stepNum : step + 1
    })
  }
&nbsp;
  render () {
    const { step } = this.state
&nbsp;
    return (
      &lt;List
        {...this.props}
        step={step}
        onNextStep={this.onNextStep}
      /&gt;
    )
  }
}
&nbsp;
export default ListContainer</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:29:52 GMT-0500 (EST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
