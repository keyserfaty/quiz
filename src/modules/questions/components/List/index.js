import React from 'react'

import List from './List'

class ListContainer extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      step: 0,
      responses: []
    }
  }

  onNextStep = () => {
    const { step } = this.state

    this.setState({
      step
    })
  }

  onAddResponse = response => {
    const { responses } = this.state

    this.setState({
      responses: [
        ...responses,
        response
      ]
    })
  }

  render () {
    return (
      <List
        {...this.props}
        onAddResponse={this.onAddResponse}
        onNextStep={this.onNextStep}
      />
    )
  }
}

export default ListContainer