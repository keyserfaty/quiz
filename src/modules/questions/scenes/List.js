import { connect } from 'react-redux'
import { withHooks } from '../../../hooks/withHooks'

import * as actions from '../actions'
import * as selectors from '../selectors'

import List from '../components/List'

const mapStateToProps = state => ({
  questions: selectors.questions(state),
  status: selectors.status(state)
})

const mapDispatchToProps = (dispatch, props) => {
  const fetchQuestions = () => dispatch({ type: actions.fetchAll })

  return {
    onMount: () => fetchQuestions(),
    onfetchQuestions: () => fetchQuestions()
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withHooks(List))
