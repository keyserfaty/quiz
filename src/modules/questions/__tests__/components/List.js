import React from 'react'
import List from '../../components/List/index'

import renderer from 'react-test-renderer'

test('Should return a List element', () => {
  const tree = renderer.create(<List />).toJSON()
  expect(tree).toMatchSnapshot()
})