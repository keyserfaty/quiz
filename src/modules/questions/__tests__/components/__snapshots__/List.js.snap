// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return a List element 1`] = `
<View
  style={Object {}}
>
  <View
    style={Object {}}
  >
    <Text>
      Welcome to the Trivia Challenge!
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text>
      You will be presented with 10 true and false questions.
    </Text>
    <Text>
      Can you score 100%?
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 50,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 15,
          "width": 235,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins-Regular",
              "fontSize": 18,
              "letterSpacing": 2,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(55,175,255,0.87)",
            },
            undefined,
          ]
        }
      >
        BEGIN
      </Text>
    </View>
  </View>
</View>
`;
